File: Sources/CountriesFlagsObjC/source/UIImage+Bahamas.m
//
//  UIImage+Bahamas.m
//  CountriesFlags
//

#import "UIImage+Bahamas.h"

#import "UIBezierPath+Extension.h"

@implementation UIImage (Bahamas)

+(UIImage * _Nullable) flagBahamasWithSize:(CGSize)size {
    UIColor* colorAquamarine = [UIColor colorWithRed:0.0f
                                               green:169.0f/255.0f
                                                blue:206.0f/255.0f
                                               alpha:1.0f];
    UIColor* colorYellow = [UIColor colorWithRed:1.0f
                                           green:199.0f/255.0f
                                            blue:44.0f/255.0f
                                           alpha:1.0f];
    UIColor* colorBlack = [UIColor blackColor];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorAquamarine setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorYellow setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 3.0f, size.width, size.height / 3.0f));
    
    [colorBlack setFill];
    [[UIBezierPath triangleInSize: size] fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Bahamas.swift
//
//  UIImage+Bahamas.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_the_Bahamas "wikipedia"
     
     Generates the flag of Bahamas.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Bahamas.
     */
    internal static func flagBahamas(with size: CGSize) -> UIImage? {
        let colorAquamarine = UIColor(red: 0.0,
                                      green: 169.0 / 255.0,
                                      blue: 206.0 / 255.0,
                                      alpha: 1.0)
        let colorYellow = UIColor(red: 1.0,
                                  green: 199.0 / 255.0,
                                  blue: 44.0 / 255.0,
                                  alpha: 1.0)
        let colorBlack = UIColor.black

        UIGraphicsBeginImageContextWithOptions(size, true, 0.0)

        colorAquamarine.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorYellow.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0, y: size.height / 3.0),
                          size: CGSize(width: size.width, height: size.height / 3.0)))

        colorBlack.setFill()
        UIBezierPath.triangle(in: size, type: .leftEquilateral).fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
